Compilation Error: samples/baby_ecs.cc

**Describe the bug**
Error happened when compiling samples/baby_ecs.cc.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to project root directory.
2. Make an empty directory called `build` (if it exists, remove its content).
3. `$ cd build/`
4. `$ CC=gcc CXX=g++ cmake ..` or `$ CC=clang CXX=clang++ cmake ..`
5. `$ make`
6. See error.

**Expected behavior**
No compilation error should occur.

**Suggestion**
Use uint32_t directly instead of std::uint32_t.

**Environment:**
 - OS: `Ubuntu 24.04 noble`
 - Kernel: `x86_64 Linux 6.8.0-47-generic`
 - Shell: `zsh 5.9`
 - GCC/G++: target `x86_64-linux-gnu`, version `(Ubuntu 13.2.0-23ubuntu4) 13.2.0`
 - Clang/Clang++: target `x86_64-pc-linux-gnu`, version `18.1.3 (1ubuntu1)`

**Shell Output**
***Case 1***: `$ CC=gcc CXX=g++ cmake ..`
```
$ CC=gcc CXX=g++ cmake .. 
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Configuring done (2.4s)
-- Generating done (0.0s)
-- Build files have been written to: /home/jerry/workplace/spargel/build
$ make      
[ 16%] Building CXX object engine/CMakeFiles/SpargelEngine.dir/ecs/entity_manager.cc.o
[ 33%] Linking CXX static library libSpargelEngine.a
[ 33%] Built target SpargelEngine
[ 50%] Building C object samples/CMakeFiles/x11_window.dir/x11_window.c.o
[ 66%] Linking C executable x11_window
[ 66%] Built target x11_window
[ 83%] Building CXX object samples/CMakeFiles/baby_ecs.dir/baby_ecs.cc.o
/home/jerry/workplace/spargel/samples/baby_ecs.cc:97:8: error: ‘uint32_t’ in namespace ‘std’ does not name a type
   97 |   std::uint32_t* sparse;
      |        ^~~~~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc:99:8: error: ‘uint32_t’ in namespace ‘std’ does not name a type
   99 |   std::uint32_t* dense;
      |        ^~~~~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In constructor ‘SparseSet::SparseSet()’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:43:5: error: ‘sparse’ was not declared in this scope; did you mean ‘SparseSet’?
   43 |     sparse =
      |     ^~~~~~
      |     SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc:46:5: error: ‘dense’ was not declared in this scope
   46 |     dense =
      |     ^~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In destructor ‘SparseSet::~SparseSet()’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:54:10: error: ‘sparse’ was not declared in this scope; did you mean ‘SparseSet’?
   54 |     free(sparse);
      |          ^~~~~~
      |          SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc:55:10: error: ‘dense’ was not declared in this scope
   55 |     free(dense);
      |          ^~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In member function ‘void SparseSet::Insert(Entity)’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:68:5: error: ‘sparse’ was not declared in this scope; did you mean ‘SparseSet’?
   68 |     sparse[entity] = counter;
      |     ^~~~~~
      |     SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc:69:5: error: ‘dense’ was not declared in this scope
   69 |     dense[counter] = entity;
      |     ^~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In member function ‘bool SparseSet::Has(Entity) const’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:75:13: error: ‘sparse’ was not declared in this scope; did you mean ‘SparseSet’?
   75 |         && (sparse[entity] < counter)
      |             ^~~~~~
      |             SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc:76:13: error: ‘dense’ was not declared in this scope
   76 |         && (dense[sparse[entity]] == entity);
      |             ^~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In member function ‘void SparseSet::Delete(Entity)’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:83:15: error: ‘dense’ was not declared in this scope
   83 |     auto e2 = dense[counter - 1];
      |               ^~~~~
/home/jerry/workplace/spargel/samples/baby_ecs.cc:84:11: error: ‘sparse’ was not declared in this scope; did you mean ‘SparseSet’?
   84 |     dense[sparse[entity]] = e2;
      |           ^~~~~~
      |           SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In member function ‘void Storage<T>::Emplace(Entity, Args&& ...)’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:122:32: error: ‘struct SparseSet’ has no member named ‘sparse’; did you mean ‘SparseSet’?
  122 |     auto ptr = real_data + set.sparse[entity];
      |                                ^~~~~~
      |                                SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In member function ‘T& Storage<T>::Get(Entity)’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:127:26: error: ‘struct SparseSet’ has no member named ‘sparse’; did you mean ‘SparseSet’?
  127 |     return real_data[set.sparse[entity]];
      |                          ^~~~~~
      |                          SparseSet
/home/jerry/workplace/spargel/samples/baby_ecs.cc: In function ‘int main()’:
/home/jerry/workplace/spargel/samples/baby_ecs.cc:239:37: error: ‘struct SparseSet’ has no member named ‘dense’
  239 |     auto entity = registry.entities.dense[i];
      |                                     ^~~~~
make[2]: *** [samples/CMakeFiles/baby_ecs.dir/build.make:76: samples/CMakeFiles/baby_ecs.dir/baby_ecs.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:170: samples/CMakeFiles/baby_ecs.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```
***Case 2***: `CC=clang CXX=clang++ cmake ..`
```
$ CC=clang CXX=clang++ cmake ..
-- The C compiler identification is Clang 18.1.3
-- The CXX compiler identification is Clang 18.1.3
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Configuring done (3.3s)
-- Generating done (0.0s)
-- Build files have been written to: /home/jerry/workplace/spargel/build
$ make
[ 16%] Building CXX object engine/CMakeFiles/SpargelEngine.dir/ecs/entity_manager.cc.o
[ 33%] Linking CXX static library libSpargelEngine.a
[ 33%] Built target SpargelEngine
[ 50%] Building C object samples/CMakeFiles/x11_window.dir/x11_window.c.o
[ 66%] Linking C executable x11_window
[ 66%] Built target x11_window
[ 83%] Building CXX object samples/CMakeFiles/baby_ecs.dir/baby_ecs.cc.o
/home/jerry/workplace/spargel/samples/baby_ecs.cc:97:3: error: no type named 'uint32_t' in namespace 'std'; did you mean simply 'uint32_t'?
   97 |   std::uint32_t* sparse;
      |   ^~~~~~~~~~~~~
      |   uint32_t
/usr/include/x86_64-linux-gnu/bits/stdint-uintn.h:26:20: note: 'uint32_t' declared here
   26 | typedef __uint32_t uint32_t;
      |                    ^
/home/jerry/workplace/spargel/samples/baby_ecs.cc:99:3: error: no type named 'uint32_t' in namespace 'std'; did you mean simply 'uint32_t'?
   99 |   std::uint32_t* dense;
      |   ^~~~~~~~~~~~~
      |   uint32_t
/usr/include/x86_64-linux-gnu/bits/stdint-uintn.h:26:20: note: 'uint32_t' declared here
   26 | typedef __uint32_t uint32_t;
      |                    ^
2 errors generated.
make[2]: *** [samples/CMakeFiles/baby_ecs.dir/build.make:76: samples/CMakeFiles/baby_ecs.dir/baby_ecs.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:170: samples/CMakeFiles/baby_ecs.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/samples/baby_ecs.cc
+++ b/samples/baby_ecs.cc
@@ -94,9 +94,9 @@ struct SparseSet {
   uint32_t counter = 0;
 
   // entity -> internal index
-  std::uint32_t* sparse;
+  uint32_t* sparse;
   // internal index -> entity
-  std::uint32_t* dense;
+  uint32_t* dense;
 };
 
 struct StorageBase {

